charm/charmcraft.yaml

```diff
@@ -0,0 +1,63 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# This file configures Charmcraft.
+# See https://juju.is/docs/sdk/charmcraft-config for guidance.
+
+name: netbox
+
+type: charm
+
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+summary: NetBox.
+
+description: |
+    NetBox is the go-to solution for modeling and documenting network
+    infrastructure for thousands of organizations worldwide. As a successor
+    to legacy IPAM and DCIM applications, NetBox provides a cohesive,
+    extensive, and accessible data model for all things networked.
+
+extensions:
+  - django-framework
+
+requires:
+  postgresql:
+    interface: postgresql_client
+    limit: 1
+
+config:
+  options:
+    redis_hostname:
```

Comment:
I'd sort of like some way to note that this is a short term thing until the 12 factor project add support for the redis relation. Can you maybe include something to that effect in the description for this field?

Reply:
Added comments in all config options that they will be removed when the redis relation is supported by the 12 factor project.